src/vstest.console/CommandLine/AssemblyMetadataProvider.cs

```diff
@@ -11,30 +11,39 @@ namespace Microsoft.VisualStudio.TestPlatform.CommandLine.Processors.Utilities
     using System.Runtime.Versioning;
     using System.Text;
     using ObjectModel;
+    using TestPlatform.Utilities.Helpers;
+    using TestPlatform.Utilities.Helpers.Interfaces;
 
     internal class AssemblyMetadataProvider : IAssemblyMetadataProvider
     {
         private static AssemblyMetadataProvider instance;
+        private IFileHelper fileHelper;
 
         /// <summary>
         /// Gets the instance.
         /// </summary>
-        internal static AssemblyMetadataProvider Instance => instance ?? (instance = new AssemblyMetadataProvider());
+        internal static AssemblyMetadataProvider Instance => instance ?? (instance = new AssemblyMetadataProvider(new FileHelper()));
+
+        internal AssemblyMetadataProvider(IFileHelper fileHelper)
```

Comment:
same comment..

Reply:
This we don't expect to used by any other assembly except test assembly. If we make this public we will lose the flexibility to change the signature.